bot/exts/backend/auto_config.py

```diff
@@ -0,0 +1,182 @@
+
+        return required_config
+
+    def check(self) -> t.Union[bool, dict]:
+        """ A check that checks if the required channels, roles etc. exist or not."""
+
+        guild = self.bot.get_guild(self.guild_id)
+
+        #checking to see if the server has the required roles or not
+        roles = [x for x in self.required()['Roles'].keys() if not discord.utils.get(guild.roles, name=x)]
+
+        #checking for the required channels
+        channels = [x for x in self.required()['Roles'].keys(
+        ) if not discord.utils.get(guild.channels, name=x)]
+
+        if not channels and not roles:
+            logger.info('The check passed, the guild has all the roles and channels.!')
+            return False
+
+        elif not roles:
+            logger.info(f'The guild has the required roles.')
+            return {'Channels': channels}
+
+        elif not channels:
+            logger.info(f'The guild has the required channels.')
+            return {'Roles': roles}
+
+        else:
+            logger.info('The guild does not have either the roles nor the channels.')
+            return {'Roles': roles, 'Channels': channels}
+
+    def false_alarm(self) -> bool:
+        """Returns true if the server is alread configured but auto config is being called. """
+
+        return os.path.exists(f'{Path.cwd()}/bot/config/config.yaml') and not self.check()
+
+    async def create_(self) -> None:
+        """A function responsible for creating all the roles and channels, if they have not already been configured."""
+
+        guild = self.bot.get_guild(self.guild_id)
+
+        if not self.check():
+            #if the statement passed, this means that the user already has channels and roles
+            #configured, we just need to get the ids.
+
+            logger.info('The server already has the required roles and channels. Getting the ids!')
+
+        elif 'Roles' in self.check().keys() and 'Channels' not in self.check().keys():
+            logger.info('Creating the required roles for the server.')
+
+            try:
+                for r in self.check()['Roles']:
+                    await guild.create_role(name=r)
+
+            except Forbidden:
+                logger.warning('The bot does not have required permissions to create the roles!')
+                return
+
+            else:
+                logger.info('Successfully created all the roles and roles')
+
+        elif 'Channels' in self.check().keys() and 'Roles' not in self.check().keys():
+            logger.info('Creating the required channels for the server.')
+            try:
+                for c in self.required()['Channels']:
+                    await guild.create_text_channel(name=c)
+            except Forbidden:
+                logger.warning('The bot does not have required permissions to create the channels!')
+                return
+
+            else:
+                logger.info('Successfully created all the roles and channels!')
```

Comment:
```suggestion
                logger.info('Successfully created all the roles.')
```
Isn't this only making the channels?

Reply:
If the code reached up the `else` statement then it means that the server doesn't have the channels configured nor does it have the roles configured.